### PR TITLE
Lesson 19 zoldi razvan

### PR DESCRIPTION
**Pull request by zoldirazvan**

### CODE REVIEW

**Comment by Liitle** [comment]
please change the base branch from `master` to the lessons branches

File: lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# considering that we should handle KeyError I create a dictionary and because we should also
+#  handle IndexError I took into consideration a LIST of dictionaries
+data = [{'name': 'Lavinia', 'age': 32},
+        {'name': 'Mihai'},  # raise exception for missing key = KeyError
+        {'name': 'Andrei', 'age': 20}]
+
+
+class JustSomeExceptions:
+    def __enter__(self):
+        self.original_dictionary = data
```
Comment (Liitle) [inline lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py:47]: this is not required :smile: We only need to deal with exceptions in the `__exit__` the `__enter__` can return `None`

File: lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import contextlib
+
+#  *******************************  1st approach   **********************************
+# considering that we should handle KeyError I create a dictionary and because we should also
+#  handle IndexError I took into consideration a LIST of dictionaries
+
+# @contextlib.contextmanager
```
Comment (Liitle) [inline lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py]: is the commented code working?

**Review by Liitle — CHANGES_REQUESTED** [review]

File: lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py (hunk 2 above)
Comment (zoldirazvan) [reply lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py]: well , it should. is not?

File: lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py (hunk 1 above)
Comment (zoldirazvan) [reply lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py:47]: somehow I tried to respect syntax and I put over there also elements that are not necessary. should I modify?

File: lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py (hunk 2 above)
Comment (zoldirazvan) [reply lesson_19/zoldi_razvan/lesson_19_zoldi_razvan.py]: I've changed a little bit first part and put 2 with's to see error handling

**Review by Liitle — APPROVED** [review]

**Pull request merged**